### PR TITLE
Compound pricer

### PR DESCRIPTION
**Pull request by antoncoding**
#  Compound Pricer

## High Level Description

Add compound pricer to get price of cToken. (real cToken value in term of USD)

A compound pricer will also need to use another price to get price of underlying.
For example, the cUSDC pricer will need a USDC pricer to work, and cETH pricer will need a ETH pricer to work.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by haythem96** [comment]
In this implementation, if we wanna push cUSDC price, we need to keep pushing USDC price also using a pricer?! It will be better to modify this and use `USDC price = 1` because it is always equal to 1. We can pricer other Pricer for different cToken when we need to.

**Comment by antoncoding** [comment]
> In this implementation, if we wanna push cUSDC price, we need to keep pushing USDC price also using a pricer?! It will be better to modify this and use `USDC price = 1` because it is always equal to 1. We can pricer other Pricer for different cToken when we need to.

Yeah, or we can deploy a Pricer for USDC that always return `1e18`. I think that will be more flexible.

**Comment by haythem96** [comment]
> > In this implementation, if we wanna push cUSDC price, we need to keep pushing USDC price also using a pricer?! It will be better to modify this and use `USDC price = 1` because it is always equal to 1. We can pricer other Pricer for different cToken when we need to.
> 
> Yeah, or we can deploy a Pricer for USDC that always return `1e18`. I think that will be more flexible.

I don't think we need that, we can just use 1e18 as it is always 1e18.

**Comment by antoncoding** [comment]
> > > In this implementation, if we wanna push cUSDC price, we need to keep pushing USDC price also using a pricer?! It will be better to modify this and use `USDC price = 1` because it is always equal to 1. We can pricer other Pricer for different cToken when we need to.
> > 
> > 
> > Yeah, or we can deploy a Pricer for USDC that always return `1e18`. I think that will be more flexible.
> 
> I don't think we need that, we can just use 1e18 as it is always 1e18.

So we should set a `USDC` address in the oracle, and always return 1e18 for that? 

I also need to change the CompoundPricer to get underlying price from oracle, instead of a pricer, so we can just return 1e18 for USDC in the oracle instead of using a Pricer.

**Comment by haythem96** [comment]
> > > > In this implementation, if we wanna push cUSDC price, we need to keep pushing USDC price also using a pricer?! It will be better to modify this and use `USDC price = 1` because it is always equal to 1. We can pricer other Pricer for different cToken when we need to.
> > > 
> > > 
> > > Yeah, or we can deploy a Pricer for USDC that always return `1e18`. I think that will be more flexible.
> > 
> > 
> > I don't think we need that, we can just use 1e18 as it is always 1e18.
> 
> So we should set a `USDC` address in the oracle, and always return 1e18 for that?
> 
> I also need to change the CompoundPricer to get underlying price from oracle, instead of a pricer, so we can just return 1e18 for USDC in the oracle instead of using a Pricer.

Oh I get ur point, so ur saying having a pricer for USDC is more generic than just adding an exception in the Oracle module for that, yeah sounds good to me then we can have an USDC pricer, but it should always return 1e18 without the need for a bot to push to it.

**Review by haythem96 — APPROVED** [review]
LGTM.

File: test/compoundPricer.test.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      const newPrice = createScaledNumber(500)
+      await chainlinkPricer.setPrice(newPrice)
+      const cTokenPrice = await compoundPricer.getPrice()
+      const expectedResult = await underlyingPriceToCtokenPrice(new BigNumber(newPrice), exchangeRate, weth)
+      assert.equal(cTokenPrice.toString(), expectedResult.toString())
+    })
+    it('should revert if price is lower than 0', async () => {
+      await chainlinkPricer.setPrice('0')
+      await expectRevert(compoundPricer.getPrice(), 'CompoundPricer: underlying price is 0')
+    })
```
Comment (aparnakr) [inline test/compoundPricer.test.ts:109]: Can you add a test to see if it works for cUSDC, cDai as well? If cUSDC works, then it tests for the different decimals properly.

File: contracts/pricers/CompoundPricer.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        oracle.setExpiryPrice(address(cToken), _expiryTimestamp, cTokenPrice);
+    }
+
+    /**
+     * @dev convert underlying price to cToken price.
+     * @param _underlyingPrice price of 1 underlying token (1e6 USDC, 1e18 WETH) in USD, scled by 1e18
+     * @return net worth of 1e8 cToken in USD, scaled by 1e18.
+     */
+    function _underlyingPriceToCtokenPrice(uint256 _underlyingPrice) internal view returns (uint256) {
+        uint256 underlyingDecimals = underlying.decimals();
```
Comment (aparnakr) [inline contracts/pricers/CompoundPricer.sol:80]: this should always be a 1e18 number right?

**Review by aparnakr — APPROVED** [review]
Looks good to me, also add tests for cUSDC to check that it works for the different decimals.

**Pull request merged**